### DISCUSSION TITLE
CRC mismatch

### DISCUSSION THREAD

**Discussion by xeni1983** [digitalpetri/modbus#85] [Q&A]
I receive this error
Caused by: com.digitalpetri.modbus.exceptions.ModbusException: CRC mismatch
It is any possibility to ignore CRC?

**Comment by kevinherron** [comment]
At the moment, no, but I think this could be added as some kind of configuration option.

It can be dangerous to ignore, the packet could be incomplete and fail to subsequently parse, or it could just silently have incorrect data in it.

**Reply by kevinherron** [reply]
Okay, something that could work in https://github.com/digitalpetri/modbus/pull/88.

Usage would be something like:
```java
    ModbusClientConfig config = ...;
    ModbusRtuClientTransport transport = ...;
    
    var client = new ModbusRtuClient(config, transport) {
      @Override
      protected boolean verifyCrc16(ModbusRtuFrame frame) {
        if (!super.verifyCrc16(frame)) {
          resetFrameParser();
        }

        // ignore CRC mismatches, return true to continue processing
        return true;
      }
    };
```

**Reply by xeni1983** [reply]
Thank you. I have an equipment that has a specific CRC validation. That's why I want to bypass verify CRC

**Reply by kevinherron** [reply]
Does it not expect a different CRC on the request frames too, then? How does this work now?

**Reply by xeni1983** [reply]
It is ok now. I receive data from sensor
